flocker/node/agents/ebs.py

```diff
@@ -42,6 +47,93 @@
 MAX_ATTACH_RETRIES = 3
 
 
+class VolumeOperations(Names):
+    """
+    Supported EBS backend operations on a volume.
+    """
+    CREATE = NamedConstant()
+    ATTACH = NamedConstant()
+    DETACH = NamedConstant()
+    DESTROY = NamedConstant()
+
+
+class VolumeStates(Values):
+    """
+    Expected EBS volume states during ``VolumeOperations``.
+    """
+    EMPTY = ValueConstant('')
+    CREATING = ValueConstant(u'creating')
+    AVAILABLE = ValueConstant(u'available')
+    ATTACHING = ValueConstant(u'attaching')
+    IN_USE = ValueConstant(u'in-use')
+    DETACHING = ValueConstant(u'detaching')
+    DELETING = ValueConstant(u'deleting')
+
+
+class VolumeStateFlow(PRecord):
+    """
+    Expected EBS volume state flow during ``VolumeOperations``.
+    """
+    start_state = field(mandatory=True, type=ValueConstant)
+    transient_state = field(mandatory=True, type=ValueConstant)
+    end_state = field(mandatory=True, type=ValueConstant)
+
+    # Boolean flag to indicate if a volume state transition
+    # results in non-empty ``attach_data.device`` and
+    # ``attach_data.instance_id`` for the EBS volume.
+    has_attach_data = field(mandatory=True, type=bool)
+
+
+class VolumeStateTable(PRecord):
+    """
+    Map of volume operation to expected volume state transitions
+    and expected update to volume's ``attach_data``.
+    """
+    table = pmap_field(NamedConstant, VolumeStateFlow)
+
+
+def populate():
+    """
+    Initialize volume state table  with transitions for ``create_volume``,
+    ``attach_volume``, ``detach_volume``, ``delete_volume`` operations.
+    """
+    table = pmap()
```

Comment:
:question: This might be easier to read if you did e.g. `S = VolumeStates` and `O = VolumeOperations` at the top.